node/subsystem-util/src/inclusion_emulator/staging.rs

```diff
@@ -757,6 +767,13 @@ fn validate_against_constraints(
 		})
 	}
 
+	if candidate.commitments.upward_messages.len() > constraints.max_ump_num_per_candidate {
```

Comment:
It looks to me like this and perhaps the block above it should both be part of `fn check_modifications` and mirrored in `fn apply_modifications`

Reply:
Modifications are stacked while this limit is imposed on a single candidate.

What you propose is not hard to implement, but nothing restricts external code from applying stacked constraints on top of something, making this check incorrect.

Reply:
Ok, that's a good point and I suppose explains why it is that way.